File: CourseProjects/CourseProjects/VisualStudioProjects/DesignPatternsExamples/Visitor/Paragraph.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Visitor {
    internal class Paragraph : IVisited {
        private readonly string text;

        public void AcceptVisitor(IVisitor v) {
            v.VisitParagraphStart(this);
            v.VisitParagraphEnd();
        }

        public Paragraph(string text) {
            this.text = text;
        }

        public override string ToString() {
            return text;
        }
    }
}

File: CourseProjects/CourseProjects/JavaEclipseProjects/DesignPatternsExamples/src/patterns/visitor/Paragraph.java
package patterns.visitor;

public class Paragraph implements Visited {
	private String text; 
	public void acceptVisitor(Visitor v) {
		v.visitParagraphStart(this);
		v.visitParagraphEnd();
	}
	public Paragraph(String text) {
		this.text = text;
	}
	public String toString() {
		return text;
	}
}
